### PR TITLE
Hapi17 decorations tests

### PR DESCRIPTION
**Pull request by zemccartney**
A pile of rewritten tests

Items of particular consideration, to help sift through :)
- `'return empty object before server initialization.'` test;  should this be patched to include adding models? Currently, no models added, so result is same even if initialization
- Is modification to the state function ok? For example, modified version used in `'return empty object if no models have been added.'` (basically, switching from inspecting server ref to realm)?

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -1331,318 +1185,258 @@ describe('Schwifty', () => {
...
+            const server = await getServer(getOptions());
 
-                server.route({
-                    path: '/',
-                    method: 'get',
-                    handler: (request, reply) => {
+            server.route({
+                path: '/',
+                method: 'get',
+                handler: (request, h) => {
```
Comment (devinivy) [inline test/index.js]: You can omit `h` if you want.

Comment (devinivy) [reply test/index.js]: Same elsewhere, if it's not used.  Might even be a lint error?

File: test/index.js [hunk 2]
```diff
@@ -1331,318 +1185,258 @@ describe('Schwifty', () => {
...
 
                             const models = request.models();
                             expect(models).to.have.length(1);
                             expect(models.Movie.tableName).to.equal('Movie');
-                            reply({ ok: 'plugin' });
+                            return { ok: 'plugin' };
                         }
                     });
-                    srv.ext('onPreStart', (_, nxt) => {
+                    srv.ext('onPreStart', (_) => {
```
Comment (devinivy) [inline test/index.js]: No need for `_`– in the past it was just a placeholder to get to `next`.

Comment (devinivy) [reply test/index.js]: Same elsewhere.

**Comment by devinivy** [comment]
Some comments, but overall this looks good 👍  Your concerns above are good questions, but no problem!

**Comment by zemccartney** [comment]
@devinivy yo! thanks for those notes, that makes sense. I'll fix those jaxsons up. 🙏

**Comment by zemccartney** [comment]
Should be all set now! 🙏

**Pull request merged**